File: python3/longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        for i in range(len(strs[0])):
            for string in strs:
                if i == len(string) or string[i] != strs[0][i]:
                    return string[:i]
        return strs[0]
# @lc code=end

File: javascript/longest-common-prefix.js
/*
 * @lc app=leetcode id=14 lang=javascript
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    for (let i = 0; i < strs[0].length; i++) {
        for (let string of strs) {
            if (string[i] != strs[0][i] || i == string.length)
                return string.slice(0, i);
        }
    }
    return strs[0];
};
// @lc code=end

/*
Vertical Scanning
Time Complexity: O(n * m)
Space Complexity: O(1)
*/
